src/median_of_median.c

```diff
@@ -15,22 +15,59 @@ void swap(int *p, int *q){
 
 int quick_select(int A[], int n, int k){
   int i, j, pivot;
-
-// 真ん中の要素をピボットとする
-  pivot = A[n/2];
-  A[n/2] = A[0];
-  A[0] = pivot;
+  if(n <= 5){
+        pivot = A[0];
   for(i = j = 1; i < n; i++){
     if(A[i] <= pivot){
       swap(A+i, A+j);
       j++;
     }
   }
 
-  if(j == k+1) return pivot;
-  else if(j < k+1) return quick_select(A+j, n-j, k-j);
-  else return quick_select(A+1, j-1, k);
-}
+      if(j == k+1) return pivot;
+      else if(j < k+1) return quick_select(A+j, n-j, k-j);
+      else return quick_select(A+1, j-1, k);
+  }
+      
+      
+                   else {
+        int x;
+        x = 0;
+                                  for(i = 0;i < (n+9)/5;i++){
+                                  if(n-x >= 5){
+                               B[i] = quick_select(A+x, 5, 2);
+                                  x=x+5;
+                           }
+            else B[i] = quick_select(A+x, n-x,(n-x)/2);
+        }
+        pivot = quick_select(B, (n+4)/5, (n+4)/10);
+        int a,b;
+        for(a = 0;a < n; a++){
+            if(A[a] == pivot){
+                b = a;
+                a = n-1;
```

Comment:
```suggestion
                break;
```
これ使うといいですよ．for文から脱することができます．